cpp/diffraction/StaticStructureFactorDebye.cc

```diff
@@ -15,9 +15,42 @@
 
 namespace freud { namespace diffraction {
 
+namespace {
+//! Given the desired k_max bin center, find the upper edge of the bin.
+float k_max_center_to_upper_edge(unsigned int bins, float k_min, float k_max)
+{
+    return k_max + (k_max - k_min) / static_cast<float>(2 * (bins - 1));
+}
+
+//! Given the desired k_min bin center, find the lower edge of the bin.
+float k_min_center_to_lower_edge(unsigned int bins, float k_min, float k_max)
+{
+    return k_min - (k_max - k_min) / static_cast<float>(2 * (bins - 1));
+}
+} // namespace
+
 StaticStructureFactorDebye::StaticStructureFactorDebye(unsigned int bins, float k_max, float k_min)
-    : StaticStructureFactor(bins, k_max, k_min)
-{}
+    : StaticStructureFactor(bins, k_max_center_to_upper_edge(bins, k_min, k_max),
+                            k_min_center_to_lower_edge(bins, k_min, k_max))
+{
+    if (bins == 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires a nonzero number of bins.");
+    }
+    if (k_max <= 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires k_max to be positive.");
+    }
+    if (k_min < 0)
+    {
+        throw std::invalid_argument("StaticStructureFactorDebye requires k_min to be non-negative.");
+    }
+    if (k_max <= k_min)
+    {
+        throw std::invalid_argument(
+            "StaticStructureFactorDebye requires that k_max must be greater than k_min.");
+    }
```

Comment:
I realize this validation logic cannot go in the super constructor anymore due to shifting needed in `StaticStructureFactorDebye`, but the code itself is the same and now exists in multiple places. To remove code duplication, let's put this is a helper method/function.

Reply:
Defining validation logic in each class is consistent with the rest of freud. We do that so the error messages can be highly descriptive without needing to mess with string concatenation / string formatting through a generalized function.

If you still think this is necessary, I would defer this to a separate PR.